federation/pkg/dnsprovider/providers/coredns/rrsets.go

```diff
@@ -78,13 +77,10 @@ func (rrsets ResourceRecordSets) Get(name string) (dnsprovider.ResourceRecordSet
 		}
```

Comment:
In the [original service definition](https://github.com/coredns/coredns/blob/master/middleware/etcd/msg/service.go) there is only 1 Host field that can either be a name or an address (A or AAAA - please don't exclude AAAA from this code). So the only way to know what you got from etcd is to parse this field. TXT/MX have extra fields. 

Btw: this is adding support for etcd3? If so I would like to have support for that in upstream as well....